tests/distributions/test_continuous.py

```diff
@@ -424,6 +424,11 @@ def scipy_log_cdf(value, a, b):
             {"a": Rplus, "b": Rplus},
             scipy_log_cdf,
         )
+        check_icdf(
+            pm.Kumaraswamy,
+            {"a": Rplus, "b": Rplus},
+            lambda q, a, b: (1 - (1 - q) ** (1 / b)) ** (1 / a),
```

Comment:
Doesn't scipy have a ppf function? The point of this test is more to compare with an external reference to be sure we implemented it correctly.

Reply:
I see you mention it doesn't. In that case an alternative would be to check that for some values `exp(logcdf(dist, icdf(dist, value))) == value` 

You could add that as a util test function in `testing.py`. 

We do something similar for the consistency between logp and logcdf helper.

It's a bit more work but it will be very helpful for testing other distributions whose ppf is missing in scipy.

Reply:
No, SciPy doesn't have a ppf function for the Kumaraswamy distribution.

Sure, I will look into adding the util test function for testing when scipy doesn't have an implementation. I will get back on this if I get stuck, or if not stuck, with the next commit.

Reply:
@ricardoV94 I added a util function for testing the inverse of the icdf matches `exp logcdf`; 
However I found it is failing tests for small values of distribution parameter $b$ due to numerical issues.

Example test case where this is failing:

$a=1$, $b=0.01$, $value=0.4$
The icdf expression is:
$(1 - (1 - value) ^ \frac{1}{b}) ^ \frac{1}{a} $

For the above values for $a$ and $b$, the inner term:
$t = (1 - value) ^ \frac{1}{b}) = 0.6^{100} = 6.533186235000685e-23$

For such a small value of $t$, $1 - t$ is now represented as 1, due to numerical issues with adding 1 to such a small number. Thus the result of the ICDF expression, $(1 - t)^\frac{1}{a}$ is also 1.

This fails the self consistency test (between icdf and cdf) for small values of $b$.

Do you have any idea to make this numerically stable? Working with the logarithms of numbers doesn't make sense as there is addition happening in the ICDF expression which also causes the numerical stability issue.

The other alternative is to simply skip the tests for small values of $b$. I would like to avoid that if another way is possible.

Reply:
How are you testing? We don't want to test equality directly, but using something like `np.testing.assert_almost_equal` or something that allows for some wiggle room. 

It may make sense to compare que log of the icdf value if that's more stable

Reply:
I am already using `np.testing.assert_almost_equal`. This still doesn't work because the probability returned by the cdf of the value returned by the icdf, is $1$, instead of a value 0.4, because of the numerical issue. 

> It may make sense to compare que log of the icdf value if that's more stable.

log of the icdf does make sense, but we have to implement a function like logicdf everywhere. log(icdf) will not work as icdf already returns the less precise value, so it has to be implemented directly as logicdf.

> But yes, we don't need to test values very close to 0 or 1.

Okay, in that case I will change the domain of testing to exclude small values of $b$.

>  It would be good to check if the distributions that already have icdf pass as well.

The other continuous distributions (uniform and normal) with icdf are passing the icdf self consistency test, even with small values of $b$.

The discrete distributions had an an error which I will try to fix next:
`TypeError: ('TensorType(int64, ()) cannot store a value of dtype float64 without risking loss of precision. If you do not mind this loss, you can: 1) explicitly cast your data to int64, or 2) set "allow_input_downcast=True" when calling "function". Value: "array(0.2)"', 'Container name "value"')`

Reply:
We don't need to implement log-icdf, that's just the log of icdf which we already have.

My proposal was to compare `logcdf(dist, icdf(dist, value)) = log(value)`, if that makes sense. Instead of exponentiating the logcdf

Reply:
No that doesn't work either, I already tried it before I narrowed down to the root cause.

The numerical error starts at the output of icdf itself for very small parameters for distributions involving power terms with this parameter. This the log outside does not help.

I will proceed now by excluding very small values from the tests.

Reply:
Well actually I see that for some of the domains, domains excluding small parameters is already defined.
For example, using `Rplusbig` instead of `Rplus` for the a and b distribution parameters for the Kumaraswamy distribution already passed the test.

I could simply use these domains for the tests. Does this sound good?

Reply:
Oh while this inverse icdf -cdf self test works for continuous distributions, it doesn't work in a straightforward way for discrete distributions (except uniform).
This is because there is a band of probability for which the icdf has the same value and this this need not match the probability from the cdf.